onnxruntime/python/tools/transformers/onnx_model_t5.py

```diff
@@ -29,25 +29,460 @@ def __init__(
         num_heads: int,
         attention_mask: AttentionMask,
     ):
-        super().__init__(model, hidden_size, num_heads, attention_mask)
+        super().__init__(
+            model,
+            hidden_size,
+            num_heads,
+            attention_mask,
+            use_multi_head_attention=False,
+            search_op_types=["SkipSimplifiedLayerNormalization", "Add"],
+        )
+        self.static_kv = 1
 
-    def create_attention_node(
+    def create_mha_node(
         self,
+        query: str,
+        key: str,
+        value: str,
         mask_index: str,
-        matmul: NodeProto,
-        add: NodeProto,
+        res_pos_bias: str,
+        past_key: str,
+        past_value: str,
+        output: str,
+        present_key: str,
+        present_value: str,
         num_heads: int,
         hidden_size: int,
-        input: str,
-        output: str,
-        add_qk_str: str,
     ) -> Union[NodeProto, None]:
-        # Not implemented yet
-        return None
+
+        assert num_heads > 0
+
+        if hidden_size > 0 and (hidden_size % num_heads) != 0:
+            logger.debug(f"input hidden size {hidden_size} is not a multiple of num of heads {num_heads}")
+            return None
+
+        attention_node_name = self.model.create_node_name("MultiHeadAttention")
+        attention_inputs = [
+            query,
+            "" if key is None else key,  # key
+            "" if value is None else value,  # value
+            "",  # bias
+        ]
+        if mask_index is not None:
+            attention_inputs.append(mask_index)
+        else:
+            attention_inputs.append("")
+
+        if res_pos_bias is not None:
+            attention_inputs.append(res_pos_bias)
+        else:
+            attention_inputs.append("")
+
+        if past_key is not None:
+            assert past_value is not None
+            attention_inputs.append(past_key)
+            attention_inputs.append(past_value)
+
+        attention_outputs = [output]
+        if present_key is not None:
+            assert present_value is not None
+            attention_outputs.append(present_key)
+            attention_outputs.append(present_value)
+
+        attention_node = helper.make_node(
+            "MultiHeadAttention",
+            inputs=attention_inputs,
+            outputs=attention_outputs,
+            name=attention_node_name,
+        )
+
+        attention_node.domain = "com.microsoft"
+        attention_node.attribute.extend([helper.make_attribute("num_heads", num_heads)])
+        attention_node.attribute.extend([helper.make_attribute("scale", 1.0)])
```

Comment:
why 1.0 here?

Reply:
https://github.com/huggingface/transformers/blob/main/src/transformers/models/t5/modeling_t5.py#L530
In t5 implementation the qk is not scaled